node/core/parachains-inherent/src/lib.rs

```diff
@@ -51,7 +51,7 @@ impl ParachainsInherentDataProvider {
 
 	/// Create a new instance of the [`ParachainsInherentDataProvider`].
 	pub async fn create<C: HeaderBackend<Block>>(
-		client: &C,
+		client: Arc<C>,
```

Comment:
Most of the logic in this function should be moved into the `provide_inherent_data` function.

Reply:
Why so? I like how this is actually a separate function handing all the data gathering.

Reply:
You can still put it into a separate function. I just don't like the duplication of the InherentDataProvider. This doesn't make any sense to me.